### ISSUE TITLE
Invalid peer certificate: UnknownIssuer

### ISSUE THREAD

**Issue by moshie** [vercel/turbo#7364]
### Verify canary release

- [X] I verified that the issue exists in the latest Turborepo canary release.

### Link to code that reproduces this issue

#

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Which canary version will you have in your reproduction?

1.12.3

### Describe the Bug

When attempting to cache using remote cache we get this error:

```
WARNING  Sending fatal alert BadCertificate
 WARNING  failed to contact remote cache: Error making HTTP request: error sending request for url (https://*****/v8/artifacts/*****?teamId=****): error trying to connect: invalid peer certificate contents: invalid peer certificate: UnknownIssuer
```

We are using a corporate proxy with a self signed certificate.

The `--go-fallback` flag has been removed so we can no longer use this but reading here it was working with that: 
https://github.com/vercel/turbo/issues/6765#issuecomment-1883748615

### Expected Behavior

When turbo is run it includes the certificate when making remote calls.

### To Reproduce

Make a remote turbo cache with a corporate proxy and self-signed certificate.

**Comment by chris-olszewski** [comment]
Can you please try with the latest canary `1.12.4-canary.1`? It has #7357 which I believe should enable the use of self-signed certs.

**Comment by moshie** [comment]
@chris-olszewski Thanks can confirm that it's fixed in `1.12.4-canary.1` 👍  Thanks
Do you know when this might be released?

**Comment by legendhimself** [comment]
```
 WARNING  rustls failed to parse DER certificate UnsupportedCriticalExtension Certificate(b"0\x82\x03\xb30\x82\x02\x9b\xa0\x03\x02\x01\x02\x02\x10I\xa9\x9b\xdaKfi\xa7L\xa0<\x817\x83X\xca0\r\x06\t*\x86H\x86\xf7\r\x01\x01\r\x05\00K1$0\"\x06\x03U\x04\x03\x13\x1bRazer Chroma SDK Local Cert1\x0f0\r\x06\x03U\x04\x0b\x13\x06Chroma1\x120\x10\x06\x03U\x04\n\x13\tRazer Inc0\x1e\x17\r240102144141Z\x17\r250102144141Z0K1$0\"\x06\x03U\x04\x03\x13\x1bRazer Chroma SDK Local Cert1\x0f0\r\x06\x03U\x04\x0b\x13\x06Chroma1\x120\x10\x06\x03U\x04\n\x13\tRazer Inc0\x82\x01\"0\r\x06\t*\x86H\x86\xf7\r\x01\x01\x01\x05\0\x03\x82\x01\x0f\00\x82\x01\n\x02\x82\x01\x01\0\xd2\xcfAr*\xa3pE\x969O\xfa]>\x8e\x03\x83-uC\xc2\x017\x1a\xe7\xdd\x15B\xf6\x80\xf5f\xb2}\x95\x97\x9b\xe7\xacFV\xfd\xe6\xd5\x8c`\x8b\x1a4}J\xf9\x0bg&\x180\x04dG{\xfe\x01\xfc*\xae\x12<\xf2\xd3\x89\x99\x02x\x0cS\x8c\x19\xe3\xa0\n\xb8\x10\xf0\xbcpZ\xa9\xa9Jx\xc5&\xcb\x1b\xec\xf3\x1f\x9b\x89\xe3c\xe1Z\xff|\xfe=t\x16\xec\x85K\xfe\xe1\x13\xb8Ty{\xf1\x9a8\x9frS\xa7\xa1pP\r,:\x1a\x87\xdcI\xc9\x7fX\xe6\x93OzBm_6\x85\xfa\x03\xab=\x14\x89$z) \x06\x95\xa0\x96\xda\xb2\x92\x88\xa6\xb7Q\x97er\n\x86\x1al^\xf6\xf6E\xf0\xdd\x80\xed/\xd7\x9c\xe2K\xb6C.\xf5\xe1\x07\xfa\x8dS\xa264y\x9b\x9f[\xd2\xda\xbd\xfb\xac\x02c\x9c\xf7\xe8\x94\xd2v\x18y\x86\xd1\xa1\xb0\xd5\x19-9\x1e\x07\x9e\xe9|f\xb6\xa0n\x08\xf0\x99\xbf\xed\xc5\xa9\xd68\xb5w\xee\xe7\xa6mA\xffi\x02\x03\x01\0\x01\xa3\x81\x920\x81\x8f0\x1d\x06\x03U\x1d\x0e\x04\x16\x04\x14\x8f\x8f\x95O\xbd\n\xf5\xdd\xed\x0b?\xb7\xfbt\x9cF\x98ypk0\x1f\x06\x03U\x1d#\x04\x180\x16\x80\x14\x8f\x8f\x95O\xbd\n\xf5\xdd\xed\x0b?\xb7\xfbt\x9cF\x98ypk0\x0f\x06\x03U\x1d\x13\x01\x01\xff\x04\x050\x03\x01\x01\xff0\x0e\x06\x03U\x1d\x0f\x01\x01\xff\x04\x04\x03\x02\x01\x060\x14\x06\x03U\x1d\x04\x01\x01\xff\x04\n0\x080\x020\0\x03\x02\x02\x040\x16\x06\x03U\x1d%\x01\x01\xff\x04\x0c0\n\x06\x08+\x06\x01\x05\x05\x07\x03\x010\r\x06\t*\x86H\x86\xf7\r\x01\x01\r\x05\0\x03\x82\x01\x01\02\xbf\x7f\xd3W0W\xa1\xef\x87\x98\xca\xb3h\xe0\x95b\xd8)\x9d\x95DS0y\xf2D\xc9b\xf9\xef\x9bs\xfd\xa2pW\xf9\xfb^\x03uFs\xb1\xe6\xbd\xbe\xe1>\xf7\xac-\xe1\xdb\xf9YT\xd5\xc0vA\x99\xcf`\xf7\x7fw\xd6\xf3\xba9\x83**m\xcd\xfc|\xa3\x0c\x85\xe9U\x96\xac\xe1\x96\x9a&\x96\xcc\xc9\xb1\xc7\x97M\xe7\x18C\xa1w}#\x93/\x80\x1b\xb1\x1a\xb40\x83\xa0p\x95\x8cYd\xad\x1af\xec\xa2\xc5\xe4\xe7\xf0z\x14\x9e\x19\xfbO\xee\xfb\xd5\xda\x91\xb7\x9d\xe0\x80z\xaa\x92t?[f\x08\xd9J$\xf5\x17\x8d\xa5\x9d\xd5\0Y\x97\x8721Rg\x0c<\xe7\xab0raGp\r\xdc\xe6\x0fX\x1eHh\x81?+R\xb9\x93 \xccO\xc8\xe5\xefO\xc3y#\x8d2\xde4\x99\xee*E\x9b\x84\x985.`\xb3T\xbc\xad\xe1*m\xa9\xbcL\xa0\t\x19\xe61\xaa\xe3\x97)~\\g\xa1$\xea\xb9+\x9c\xd2\0\x02\x8b\x9f\xec\x13\xa9\xed6\0rZ")
 WARNING  rustls failed to parse DER certificate UnsupportedCriticalExtension Certificate(b"0\x82\x03\xb30\x82\x02\x9b\xa0\x03\x02\x01\x02\x02\x10I\xa9\x9b\xdaKfi\xa7L\xa0<\x817\x83X\xca0\r\x06\t*\x86H\x86\xf7\r\x01\x01\r\x05\00K1$0\"\x06\x03U\x04\x03\x13\x1bRazer Chroma SDK Local Cert1\x0f0\r\x06\x03U\x04\x0b\x13\x06Chroma1\x120\x10\x06\x03U\x04\n\x13\tRazer Inc0\x1e\x17\r240102144141Z\x17\r250102144141Z0K1$0\"\x06\x03U\x04\x03\x13\x1bRazer Chroma SDK Local Cert1\x0f0\r\x06\x03U\x04\x0b\x13\x06Chroma1\x120\x10\x06\x03U\x04\n\x13\tRazer Inc0\x82\x01\"0\r\x06\t*\x86H\x86\xf7\r\x01\x01\x01\x05\0\x03\x82\x01\x0f\00\x82\x01\n\x02\x82\x01\x01\0\xd2\xcfAr*\xa3pE\x969O\xfa]>\x8e\x03\x83-uC\xc2\x017\x1a\xe7\xdd\x15B\xf6\x80\xf5f\xb2}\x95\x97\x9b\xe7\xacFV\xfd\xe6\xd5\x8c`\x8b\x1a4}J\xf9\x0bg&\x180\x04dG{\xfe\x01\xfc*\xae\x12<\xf2\xd3\x89\x99\x02x\x0cS\x8c\x19\xe3\xa0\n\xb8\x10\xf0\xbcpZ\xa9\xa9Jx\xc5&\xcb\x1b\xec\xf3\x1f\x9b\x89\xe3c\xe1Z\xff|\xfe=t\x16\xec\x85K\xfe\xe1\x13\xb8Ty{\xf1\x9a8\x9frS\xa7\xa1pP\r,:\x1a\x87\xdcI\xc9\x7fX\xe6\x93OzBm_6\x85\xfa\x03\xab=\x14\x89$z) \x06\x95\xa0\x96\xda\xb2\x92\x88\xa6\xb7Q\x97er\n\x86\x1al^\xf6\xf6E\xf0\xdd\x80\xed/\xd7\x9c\xe2K\xb6C.\xf5\xe1\x07\xfa\x8dS\xa264y\x9b\x9f[\xd2\xda\xbd\xfb\xac\x02c\x9c\xf7\xe8\x94\xd2v\x18y\x86\xd1\xa1\xb0\xd5\x19-9\x1e\x07\x9e\xe9|f\xb6\xa0n\x08\xf0\x99\xbf\xed\xc5\xa9\xd68\xb5w\xee\xe7\xa6mA\xffi\x02\x03\x01\0\x01\xa3\x81\x920\x81\x8f0\x1d\x06\x03U\x1d\x0e\x04\x16\x04\x14\x8f\x8f\x95O\xbd\n\xf5\xdd\xed\x0b?\xb7\xfbt\x9cF\x98ypk0\x1f\x06\x03U\x1d#\x04\x180\x16\x80\x14\x8f\x8f\x95O\xbd\n\xf5\xdd\xed\x0b?\xb7\xfbt\x9cF\x98ypk0\x0f\x06\x03U\x1d\x13\x01\x01\xff\x04\x050\x03\x01\x01\xff0\x0e\x06\x03U\x1d\x0f\x01\x01\xff\x04\x04\x03\x02\x01\x060\x14\x06\x03U\x1d\x04\x01\x01\xff\x04\n0\x080\x020\0\x03\x02\x02\x040\x16\x06\x03U\x1d%\x01\x01\xff\x04\x0c0\n\x06\x08+\x06\x01\x05\x05\x07\x03\x010\r\x06\t*\x86H\x86\xf7\r\x01\x01\r\x05\0\x03\x82\x01\x01\02\xbf\x7f\xd3W0W\xa1\xef\x87\x98\xca\xb3h\xe0\x95b\xd8)\x9d\x95DS0y\xf2D\xc9b\xf9\xef\x9bs\xfd\xa2pW\xf9\xfb^\x03uFs\xb1\xe6\xbd\xbe\xe1>\xf7\xac-\xe1\xdb\xf9YT\xd5\xc0vA\x99\xcf`\xf7\x7fw\xd6\xf3\xba9\x83**m\xcd\xfc|\xa3\x0c\x85\xe9U\x96\xac\xe1\x96\x9a&\x96\xcc\xc9\xb1\xc7\x97M\xe7\x18C\xa1w}#\x93/\x80\x1b\xb1\x1a\xb40\x83\xa0p\x95\x8cYd\xad\x1af\xec\xa2\xc5\xe4\xe7\xf0z\x14\x9e\x19\xfbO\xee\xfb\xd5\xda\x91\xb7\x9d\xe0\x80z\xaa\x92t?[f\x08\xd9J$\xf5\x17\x8d\xa5\x9d\xd5\0Y\x97\x8721Rg\x0c<\xe7\xab0raGp\r\xdc\xe6\x0fX\x1eHh\x81?+R\xb9\x93 \xccO\xc8\xe5\xefO\xc3y#\x8d2\xde4\x99\xee*E\x9b\x84\x985.`\xb3T\xbc\xad\xe1*m\xa9\xbcL\xa0\t\x19\xe61\xaa\xe3\x97)~\\g\xa1$\xea\xb9+\x9c\xd2\0\x02\x8b\x9f\xec\x13\xa9\xed6\0rZ")
 ```
 
 I am getting this error after this release.
